langchain-core/src/runnables/passthrough.ts

```diff
@@ -31,6 +33,66 @@ export class RunnableAssign<
       ...mapperResult,
     } as RunOutput;
   }
+
+  async *_transform(
+    generator: AsyncGenerator<RunInput>,
+    runManager?: CallbackManagerForChainRun,
+    options?: Partial<RunnableConfig>
+  ): AsyncGenerator<RunOutput> {
+    // collect mapper keys
+    const mapper_keys = this.mapper.getStepsKeys();
+    // create two input gens, one for the mapper, one for the input
+    const [forPassthrough, forMapper] = atee(generator, 2);
+    // create mapper output gen
+    const mapperOutput = this.mapper.transform(
```

Comment:
What's the purpose of the tee? Can we not just:

```typescript
      const filtered = Object.fromEntries(
        Object.entries(chunk).filter(([key]) => !mapper_keys.includes(key))
      );
      if (Object.keys(filtered).length > 0) {
        yield filtered as unknown as RunOutput;
      }
```
below along with the other chunks?

Reply:
We can't consume the same generator twice

Reply:
Oh I get it, never mind.